tracee-ebpf/tracee/network_protocols/net_packet.go

```diff
@@ -0,0 +1,63 @@
+package network_protocols
+
+import (
+	"bytes"
+	"encoding/binary"
+	"github.com/aquasecurity/tracee/pkg/external"
+	"github.com/aquasecurity/tracee/pkg/processContext"
+)
+
+type PacketMeta struct {
+	PktLen   uint32   `json:"pkt_len"`
+	IfIndex  uint32   `json:"if_index"`
+	SrcIP    [16]byte `json:"src_ip"`
+	DestIP   [16]byte `json:"dest_ip"`
+	SrcPort  uint16   `json:"src_port"`
+	DestPort uint16   `json:"dest_port"`
+	Protocol uint8    `json:"protocol"`
+	_        [3]byte  //padding
+}
+
+func netPacketProtocolHandler(buffer *bytes.Buffer, evtMeta EventMeta, ctx processContext.ProcessCtx, eventName string) (external.Event, PacketMeta) {
+	var evt external.Event
+	packet, err := ParseNetPacketMetaData(buffer)
+	if err != nil {
+		return evt, packet
+	}
+	evt = CreateNetEvent(evtMeta, ctx)
+	CreateNetPacketMetaArgs(&evt, packet)
+	evt.EventName = eventName
+	return evt, packet
+}
+
+// parsing the PacketMeta struct from bytes.buffer
+func ParseNetPacketMetaData(payload *bytes.Buffer) (PacketMeta, error) {
+	var pktMetaData PacketMeta
+	err := binary.Read(payload, binary.LittleEndian, &pktMetaData)
+	if err != nil {
+		return pktMetaData, err
+	}
+	return pktMetaData, nil
+}
+
+//takes the packet metadata and create argument array with that data
+func createNetPacketMetadataArg(packet PacketMeta) []external.Argument {
+	eventArgs := make([]external.Argument, 0, 0)
+	arg := external.PktMeta{}
```

Comment:
same here: you can use `external.PktMeta` for the first arg (src ip, src port, dst ip, dst port, protocol).

Reply:
That's what I do, don't I?

Reply:
your'e right, my bad :)